iceoryx_hoofs/test/moduletests/test_vocabulary_semantic_string.cpp

```diff
@@ -0,0 +1,367 @@
+    {
+        for (auto& invalid_value : TestValues<SutType>::INVALID_CHARACTER_VALUES)
+        {
+            auto sut = SutType::create(string<SutType::capacity()>(TruncateToCapacity, value.c_str()));
+            ASSERT_THAT(sut.has_error(), Eq(false));
+
+            EXPECT_THAT(sut->append(string<SutType::capacity()>(TruncateToCapacity, invalid_value.c_str())).has_error(),
+                        Eq(true));
+            EXPECT_THAT(sut->size(), value.size());
+            EXPECT_THAT(sut->capacity(), Eq(TestValues<SutType>::CAPACITY));
+
+            EXPECT_THAT(sut->as_string().c_str(), StrEq(value));
+        }
+    }
+}
+
+TYPED_TEST(SemanticString_test, AppendTooLongContentToValidStringFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "b8616fbf-601d-43b9-b4a3-f6b96acdf555");
+    using SutType = typename TestFixture::SutType;
+
+    for (auto& value : TestValues<SutType>::VALID_VALUES)
+    {
+        for (auto& invalid_value : TestValues<SutType>::TOO_LONG_CONTENT_VALUES)
+        {
+            auto sut = SutType::create(string<SutType::capacity()>(TruncateToCapacity, value.c_str()));
+            ASSERT_THAT(sut.has_error(), Eq(false));
+
+            EXPECT_THAT(sut->append(string<SutType::capacity()>(TruncateToCapacity, invalid_value.c_str())).has_error(),
+                        Eq(true));
+            EXPECT_THAT(sut->size(), Eq(value.size()));
+            EXPECT_THAT(sut->capacity(), Eq(TestValues<SutType>::CAPACITY));
+
+            EXPECT_THAT(sut->as_string().c_str(), StrEq(value));
+        }
+    }
+}
+
+TYPED_TEST(SemanticString_test, InsertValidContentToValidStringWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "56ea499f-5ac3-4ffe-abea-b56194cfd728");
+    using SutType = typename TestFixture::SutType;
+
+    for (auto& value : TestValues<SutType>::VALID_VALUES)
+    {
+        for (auto& add_value : TestValues<SutType>::VALID_VALUES)
+        {
+            for (uint64_t insert_position = 0; insert_position < value.size(); ++insert_position)
+            {
+                auto sut = SutType::create(string<SutType::capacity()>(TruncateToCapacity, value.c_str()));
+                ASSERT_THAT(sut.has_error(), Eq(false));
+
+                EXPECT_THAT(sut->insert(insert_position,
+                                        string<SutType::capacity()>(TruncateToCapacity, add_value.c_str()),
+                                        add_value.size())
+                                .has_error(),
+                            Eq(false));
+
+
+                EXPECT_THAT(sut->size(), Eq(value.size() + add_value.size()));
+                EXPECT_THAT(sut->capacity(), Eq(TestValues<SutType>::CAPACITY));
+
+                auto result = value;
+                result.insert(insert_position, add_value);
+                EXPECT_THAT(sut->as_string().c_str(), StrEq(result));
+            }
+        }
+    }
+}
+
+TYPED_TEST(SemanticString_test, InsertInvalidContentToValidStringFails)
```

Comment:
This is similar to `AppendInvalidContentToValidStringFails` with `InvalidContent` in the test name and `INVALID_CHARACTER` for test values. Since `append` uses `insert` I don't think it is required to duplicate the test but only to adjust the names.